waltz_pileupmatrices_2.0/waltz_pileupmatrices_2-0.cwl

```diff
@@ -0,0 +1,87 @@
+class: CommandLineTool
+cwlVersion: v1.0
+id: waltz_pileupmatrices
+baseCommand:
+  - java
+inputs:
+  - id: bam
+    type: File?
+    inputBinding:
+      position: 11
+    secondaryFiles:
+      - ^.bai
+  - id: referece_fasta
+    type: File?
+    inputBinding:
+      position: 12
+    secondaryFiles:
+      - .fai
+  - default: 20
+    id: min_map_quality
+    type: int
+    inputBinding:
+      position: 10
+  - id: memory_per_job
+    type: int?
+  - id: memory_overhead
+    type: int?
+  - id: number_of_threads
+    type: int?
+  - id: bed_file
+    type: File?
+    inputBinding:
+      position: 13
+outputs:
+  - id: pileup
+    type: File?
+    outputBinding:
+      glob: '*-pileup.txt'
+  - id: pileup_without_duplicates
+    type: File?
+    outputBinding:
+      glob: '*-pileup-without-duplicates.txt'
+  - id: intervals
+    type: File?
+    outputBinding:
+      glob: '*-intervals.txt'
+  - id: intervals_without_duplicates
+    type: File?
+    outputBinding:
+      glob: '*-intervals-without-duplicates.txt'
+label: waltz_pileupmetrices
+arguments:
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: '-server'
+  - position: 1
+    prefix: ''
+    separate: false
+    valueFrom: "${\n  if(inputs.memory_per_job && inputs.memory_overhead) {\n   \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xms\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xms\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if (inputs.memory_per_job && !inputs.memory_overhead){\n    \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xms\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xms\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if(!inputs.memory_per_job && inputs.memory_overhead){\n    \n    return \"-Xms4G\"\n  }\n  else {\n    \n  \treturn \"-Xms4G\"\n  }\n}"
+  - position: 2
+    prefix: ''
+    separate: false
+    valueFrom: "${\n  if(inputs.memory_per_job && inputs.memory_overhead) {\n   \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if (inputs.memory_per_job && !inputs.memory_overhead){\n    \n    if(inputs.memory_per_job % 1000 == 0) {\n    \t\n      return \"-Xmx\" + (inputs.memory_per_job/1000).toString() + \"G\"\n    }\n    else {\n      \n      return \"-Xmx\" + Math.floor((inputs.memory_per_job/1000)).toString() + \"G\" \n    }\n  }\n  else if(!inputs.memory_per_job && inputs.memory_overhead){\n    \n    return \"-Xmx4G\"\n  }\n  else {\n    \n  \treturn \"-Xmx4G\"\n  }\n}"
+  - position: 3
+    prefix: ''
```

Comment:
Can we remove the `prefix: ''` from all

Reply:
The `prefix: ''`is added by default by **Rabix composer** whenever there is a `valueFrom` value. The `separate: false` is needed for the CWL to be correctly compiled with **Cwltool** executer. Since the plan was to write CWLs within Rabix composer, I am going to keep it this way to prevent any issues.